### ISSUE TITLE
Parsing via "read" might have a bug

### ISSUE THREAD

**Issue by saurabhnanda** [PaulJohnson/Haskell-Decimal#17]
Notice what the space between list elements does to the parsing results...

```
Prelude Data.Decimal> read "[1, 2]" :: [Int]
[1,2]
Prelude Data.Decimal> read "[1,2]" :: [Int]
[1,2]
Prelude Data.Decimal> read "[1,      2]" :: [Int]
[1,2]
Prelude Data.Decimal> read "[1.0,2.1]" :: [Decimal]
[1.0,2.1]
Prelude Data.Decimal> read "[1.0, 2.1]" :: [Decimal]
*** Exception: Prelude.read: no parse
```

**Comment by PaulJohnson** [comment]
Yup, thats a bug. It also affects bare values with leading spaces.

**Comment by PaulJohnson** [comment]
Fixed in 0.5.1.
